// Code generated by smithy-swift-codegen. DO NOT EDIT!

import ClientRuntime

/// Amazon Macie
public protocol Macie2ClientProtocol {
    /// Accepts an Amazon Macie membership invitation that was received from a specific account.
    ///
    /// - Parameter AcceptInvitationInput : [no documentation found]
    ///
    /// - Returns: `AcceptInvitationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func acceptInvitation(input: AcceptInvitationInput) async throws -> AcceptInvitationOutput
    /// Retrieves information about one or more custom data identifiers.
    ///
    /// - Parameter BatchGetCustomDataIdentifiersInput : [no documentation found]
    ///
    /// - Returns: `BatchGetCustomDataIdentifiersOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func batchGetCustomDataIdentifiers(input: BatchGetCustomDataIdentifiersInput) async throws -> BatchGetCustomDataIdentifiersOutput
    /// Creates and defines the settings for an allow list.
    ///
    /// - Parameter CreateAllowListInput : [no documentation found]
    ///
    /// - Returns: `CreateAllowListOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createAllowList(input: CreateAllowListInput) async throws -> CreateAllowListOutput
    /// Creates and defines the settings for a classification job.
    ///
    /// - Parameter CreateClassificationJobInput : [no documentation found]
    ///
    /// - Returns: `CreateClassificationJobOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createClassificationJob(input: CreateClassificationJobInput) async throws -> CreateClassificationJobOutput
    /// Creates and defines the criteria and other settings for a custom data identifier.
    ///
    /// - Parameter CreateCustomDataIdentifierInput : [no documentation found]
    ///
    /// - Returns: `CreateCustomDataIdentifierOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createCustomDataIdentifier(input: CreateCustomDataIdentifierInput) async throws -> CreateCustomDataIdentifierOutput
    /// Creates and defines the criteria and other settings for a findings filter.
    ///
    /// - Parameter CreateFindingsFilterInput : [no documentation found]
    ///
    /// - Returns: `CreateFindingsFilterOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createFindingsFilter(input: CreateFindingsFilterInput) async throws -> CreateFindingsFilterOutput
    /// Sends an Amazon Macie membership invitation to one or more accounts.
    ///
    /// - Parameter CreateInvitationsInput : [no documentation found]
    ///
    /// - Returns: `CreateInvitationsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createInvitations(input: CreateInvitationsInput) async throws -> CreateInvitationsOutput
    /// Associates an account with an Amazon Macie administrator account.
    ///
    /// - Parameter CreateMemberInput : [no documentation found]
    ///
    /// - Returns: `CreateMemberOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createMember(input: CreateMemberInput) async throws -> CreateMemberOutput
    /// Creates sample findings.
    ///
    /// - Parameter CreateSampleFindingsInput : [no documentation found]
    ///
    /// - Returns: `CreateSampleFindingsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func createSampleFindings(input: CreateSampleFindingsInput) async throws -> CreateSampleFindingsOutput
    /// Declines Amazon Macie membership invitations that were received from specific accounts.
    ///
    /// - Parameter DeclineInvitationsInput : [no documentation found]
    ///
    /// - Returns: `DeclineInvitationsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func declineInvitations(input: DeclineInvitationsInput) async throws -> DeclineInvitationsOutput
    /// Deletes an allow list.
    ///
    /// - Parameter DeleteAllowListInput : [no documentation found]
    ///
    /// - Returns: `DeleteAllowListOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func deleteAllowList(input: DeleteAllowListInput) async throws -> DeleteAllowListOutput
    /// Soft deletes a custom data identifier.
    ///
    /// - Parameter DeleteCustomDataIdentifierInput : [no documentation found]
    ///
    /// - Returns: `DeleteCustomDataIdentifierOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func deleteCustomDataIdentifier(input: DeleteCustomDataIdentifierInput) async throws -> DeleteCustomDataIdentifierOutput
    /// Deletes a findings filter.
    ///
    /// - Parameter DeleteFindingsFilterInput : [no documentation found]
    ///
    /// - Returns: `DeleteFindingsFilterOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func deleteFindingsFilter(input: DeleteFindingsFilterInput) async throws -> DeleteFindingsFilterOutput
    /// Deletes Amazon Macie membership invitations that were received from specific accounts.
    ///
    /// - Parameter DeleteInvitationsInput : [no documentation found]
    ///
    /// - Returns: `DeleteInvitationsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func deleteInvitations(input: DeleteInvitationsInput) async throws -> DeleteInvitationsOutput
    /// Deletes the association between an Amazon Macie administrator account and an account.
    ///
    /// - Parameter DeleteMemberInput : [no documentation found]
    ///
    /// - Returns: `DeleteMemberOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func deleteMember(input: DeleteMemberInput) async throws -> DeleteMemberOutput
    /// Retrieves (queries) statistical data and other information about one or more S3 buckets that Amazon Macie monitors and analyzes for an account.
    ///
    /// - Parameter DescribeBucketsInput : [no documentation found]
    ///
    /// - Returns: `DescribeBucketsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func describeBuckets(input: DescribeBucketsInput) async throws -> DescribeBucketsOutput
    /// Retrieves the status and settings for a classification job.
    ///
    /// - Parameter DescribeClassificationJobInput : [no documentation found]
    ///
    /// - Returns: `DescribeClassificationJobOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func describeClassificationJob(input: DescribeClassificationJobInput) async throws -> DescribeClassificationJobOutput
    /// Retrieves the Amazon Macie configuration settings for an organization in Organizations.
    ///
    /// - Parameter DescribeOrganizationConfigurationInput : [no documentation found]
    ///
    /// - Returns: `DescribeOrganizationConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func describeOrganizationConfiguration(input: DescribeOrganizationConfigurationInput) async throws -> DescribeOrganizationConfigurationOutput
    /// Disables Amazon Macie and deletes all settings and resources for a Macie account.
    ///
    /// - Parameter DisableMacieInput : [no documentation found]
    ///
    /// - Returns: `DisableMacieOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func disableMacie(input: DisableMacieInput) async throws -> DisableMacieOutput
    /// Disables an account as the delegated Amazon Macie administrator account for an organization in Organizations.
    ///
    /// - Parameter DisableOrganizationAdminAccountInput : [no documentation found]
    ///
    /// - Returns: `DisableOrganizationAdminAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func disableOrganizationAdminAccount(input: DisableOrganizationAdminAccountInput) async throws -> DisableOrganizationAdminAccountOutput
    /// Disassociates a member account from its Amazon Macie administrator account.
    ///
    /// - Parameter DisassociateFromAdministratorAccountInput : [no documentation found]
    ///
    /// - Returns: `DisassociateFromAdministratorAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func disassociateFromAdministratorAccount(input: DisassociateFromAdministratorAccountInput) async throws -> DisassociateFromAdministratorAccountOutput
    /// (Deprecated) Disassociates a member account from its Amazon Macie administrator account. This operation has been replaced by the DisassociateFromAdministratorAccount operation.
    ///
    /// - Parameter DisassociateFromMasterAccountInput : [no documentation found]
    ///
    /// - Returns: `DisassociateFromMasterAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func disassociateFromMasterAccount(input: DisassociateFromMasterAccountInput) async throws -> DisassociateFromMasterAccountOutput
    /// Disassociates an Amazon Macie administrator account from a member account.
    ///
    /// - Parameter DisassociateMemberInput : [no documentation found]
    ///
    /// - Returns: `DisassociateMemberOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func disassociateMember(input: DisassociateMemberInput) async throws -> DisassociateMemberOutput
    /// Enables Amazon Macie and specifies the configuration settings for a Macie account.
    ///
    /// - Parameter EnableMacieInput : [no documentation found]
    ///
    /// - Returns: `EnableMacieOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func enableMacie(input: EnableMacieInput) async throws -> EnableMacieOutput
    /// Designates an account as the delegated Amazon Macie administrator account for an organization in Organizations.
    ///
    /// - Parameter EnableOrganizationAdminAccountInput : [no documentation found]
    ///
    /// - Returns: `EnableOrganizationAdminAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func enableOrganizationAdminAccount(input: EnableOrganizationAdminAccountInput) async throws -> EnableOrganizationAdminAccountOutput
    /// Retrieves information about the Amazon Macie administrator account for an account.
    ///
    /// - Parameter GetAdministratorAccountInput : [no documentation found]
    ///
    /// - Returns: `GetAdministratorAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getAdministratorAccount(input: GetAdministratorAccountInput) async throws -> GetAdministratorAccountOutput
    /// Retrieves the settings and status of an allow list.
    ///
    /// - Parameter GetAllowListInput : [no documentation found]
    ///
    /// - Returns: `GetAllowListOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getAllowList(input: GetAllowListInput) async throws -> GetAllowListOutput
    /// Retrieves the configuration settings and status of automated sensitive data discovery for an account.
    ///
    /// - Parameter GetAutomatedDiscoveryConfigurationInput : [no documentation found]
    ///
    /// - Returns: `GetAutomatedDiscoveryConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getAutomatedDiscoveryConfiguration(input: GetAutomatedDiscoveryConfigurationInput) async throws -> GetAutomatedDiscoveryConfigurationOutput
    /// Retrieves (queries) aggregated statistical data about all the S3 buckets that Amazon Macie monitors and analyzes for an account.
    ///
    /// - Parameter GetBucketStatisticsInput : [no documentation found]
    ///
    /// - Returns: `GetBucketStatisticsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getBucketStatistics(input: GetBucketStatisticsInput) async throws -> GetBucketStatisticsOutput
    /// Retrieves the configuration settings for storing data classification results.
    ///
    /// - Parameter GetClassificationExportConfigurationInput : [no documentation found]
    ///
    /// - Returns: `GetClassificationExportConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getClassificationExportConfiguration(input: GetClassificationExportConfigurationInput) async throws -> GetClassificationExportConfigurationOutput
    /// Retrieves the classification scope settings for an account.
    ///
    /// - Parameter GetClassificationScopeInput : [no documentation found]
    ///
    /// - Returns: `GetClassificationScopeOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getClassificationScope(input: GetClassificationScopeInput) async throws -> GetClassificationScopeOutput
    /// Retrieves the criteria and other settings for a custom data identifier.
    ///
    /// - Parameter GetCustomDataIdentifierInput : [no documentation found]
    ///
    /// - Returns: `GetCustomDataIdentifierOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getCustomDataIdentifier(input: GetCustomDataIdentifierInput) async throws -> GetCustomDataIdentifierOutput
    /// Retrieves the details of one or more findings.
    ///
    /// - Parameter GetFindingsInput : [no documentation found]
    ///
    /// - Returns: `GetFindingsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getFindings(input: GetFindingsInput) async throws -> GetFindingsOutput
    /// Retrieves the criteria and other settings for a findings filter.
    ///
    /// - Parameter GetFindingsFilterInput : [no documentation found]
    ///
    /// - Returns: `GetFindingsFilterOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getFindingsFilter(input: GetFindingsFilterInput) async throws -> GetFindingsFilterOutput
    /// Retrieves the configuration settings for publishing findings to Security Hub.
    ///
    /// - Parameter GetFindingsPublicationConfigurationInput : [no documentation found]
    ///
    /// - Returns: `GetFindingsPublicationConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getFindingsPublicationConfiguration(input: GetFindingsPublicationConfigurationInput) async throws -> GetFindingsPublicationConfigurationOutput
    /// Retrieves (queries) aggregated statistical data about findings.
    ///
    /// - Parameter GetFindingStatisticsInput : [no documentation found]
    ///
    /// - Returns: `GetFindingStatisticsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getFindingStatistics(input: GetFindingStatisticsInput) async throws -> GetFindingStatisticsOutput
    /// Retrieves the count of Amazon Macie membership invitations that were received by an account.
    ///
    /// - Parameter GetInvitationsCountInput : [no documentation found]
    ///
    /// - Returns: `GetInvitationsCountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getInvitationsCount(input: GetInvitationsCountInput) async throws -> GetInvitationsCountOutput
    /// Retrieves the status and configuration settings for an Amazon Macie account.
    ///
    /// - Parameter GetMacieSessionInput : [no documentation found]
    ///
    /// - Returns: `GetMacieSessionOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getMacieSession(input: GetMacieSessionInput) async throws -> GetMacieSessionOutput
    /// (Deprecated) Retrieves information about the Amazon Macie administrator account for an account. This operation has been replaced by the GetAdministratorAccount operation.
    ///
    /// - Parameter GetMasterAccountInput : [no documentation found]
    ///
    /// - Returns: `GetMasterAccountOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getMasterAccount(input: GetMasterAccountInput) async throws -> GetMasterAccountOutput
    /// Retrieves information about an account that's associated with an Amazon Macie administrator account.
    ///
    /// - Parameter GetMemberInput : [no documentation found]
    ///
    /// - Returns: `GetMemberOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getMember(input: GetMemberInput) async throws -> GetMemberOutput
    /// Retrieves (queries) sensitive data discovery statistics and the sensitivity score for an S3 bucket.
    ///
    /// - Parameter GetResourceProfileInput : [no documentation found]
    ///
    /// - Returns: `GetResourceProfileOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getResourceProfile(input: GetResourceProfileInput) async throws -> GetResourceProfileOutput
    /// Retrieves the status and configuration settings for retrieving occurrences of sensitive data reported by findings.
    ///
    /// - Parameter GetRevealConfigurationInput : [no documentation found]
    ///
    /// - Returns: `GetRevealConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getRevealConfiguration(input: GetRevealConfigurationInput) async throws -> GetRevealConfigurationOutput
    /// Retrieves occurrences of sensitive data reported by a finding.
    ///
    /// - Parameter GetSensitiveDataOccurrencesInput : [no documentation found]
    ///
    /// - Returns: `GetSensitiveDataOccurrencesOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `UnprocessableEntityException` : Provides information about an error that occurred due to an unprocessable entity.
    func getSensitiveDataOccurrences(input: GetSensitiveDataOccurrencesInput) async throws -> GetSensitiveDataOccurrencesOutput
    /// Checks whether occurrences of sensitive data can be retrieved for a finding.
    ///
    /// - Parameter GetSensitiveDataOccurrencesAvailabilityInput : [no documentation found]
    ///
    /// - Returns: `GetSensitiveDataOccurrencesAvailabilityOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    func getSensitiveDataOccurrencesAvailability(input: GetSensitiveDataOccurrencesAvailabilityInput) async throws -> GetSensitiveDataOccurrencesAvailabilityOutput
    /// Retrieves the settings for the sensitivity inspection template for an account.
    ///
    /// - Parameter GetSensitivityInspectionTemplateInput : [no documentation found]
    ///
    /// - Returns: `GetSensitivityInspectionTemplateOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getSensitivityInspectionTemplate(input: GetSensitivityInspectionTemplateInput) async throws -> GetSensitivityInspectionTemplateOutput
    /// Retrieves (queries) quotas and aggregated usage data for one or more accounts.
    ///
    /// - Parameter GetUsageStatisticsInput : [no documentation found]
    ///
    /// - Returns: `GetUsageStatisticsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getUsageStatistics(input: GetUsageStatisticsInput) async throws -> GetUsageStatisticsOutput
    /// Retrieves (queries) aggregated usage data for an account.
    ///
    /// - Parameter GetUsageTotalsInput : [no documentation found]
    ///
    /// - Returns: `GetUsageTotalsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func getUsageTotals(input: GetUsageTotalsInput) async throws -> GetUsageTotalsOutput
    /// Retrieves a subset of information about all the allow lists for an account.
    ///
    /// - Parameter ListAllowListsInput : [no documentation found]
    ///
    /// - Returns: `ListAllowListsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listAllowLists(input: ListAllowListsInput) async throws -> ListAllowListsOutput
    /// Retrieves a subset of information about one or more classification jobs.
    ///
    /// - Parameter ListClassificationJobsInput : [no documentation found]
    ///
    /// - Returns: `ListClassificationJobsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listClassificationJobs(input: ListClassificationJobsInput) async throws -> ListClassificationJobsOutput
    /// Retrieves a subset of information about the classification scope for an account.
    ///
    /// - Parameter ListClassificationScopesInput : [no documentation found]
    ///
    /// - Returns: `ListClassificationScopesOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listClassificationScopes(input: ListClassificationScopesInput) async throws -> ListClassificationScopesOutput
    /// Retrieves a subset of information about all the custom data identifiers for an account.
    ///
    /// - Parameter ListCustomDataIdentifiersInput : [no documentation found]
    ///
    /// - Returns: `ListCustomDataIdentifiersOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listCustomDataIdentifiers(input: ListCustomDataIdentifiersInput) async throws -> ListCustomDataIdentifiersOutput
    /// Retrieves a subset of information about one or more findings.
    ///
    /// - Parameter ListFindingsInput : [no documentation found]
    ///
    /// - Returns: `ListFindingsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listFindings(input: ListFindingsInput) async throws -> ListFindingsOutput
    /// Retrieves a subset of information about all the findings filters for an account.
    ///
    /// - Parameter ListFindingsFiltersInput : [no documentation found]
    ///
    /// - Returns: `ListFindingsFiltersOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listFindingsFilters(input: ListFindingsFiltersInput) async throws -> ListFindingsFiltersOutput
    /// Retrieves information about the Amazon Macie membership invitations that were received by an account.
    ///
    /// - Parameter ListInvitationsInput : [no documentation found]
    ///
    /// - Returns: `ListInvitationsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listInvitations(input: ListInvitationsInput) async throws -> ListInvitationsOutput
    /// Retrieves information about all the managed data identifiers that Amazon Macie currently provides.
    ///
    /// - Parameter ListManagedDataIdentifiersInput : [no documentation found]
    ///
    /// - Returns: `ListManagedDataIdentifiersOutput` : [no documentation found]
    func listManagedDataIdentifiers(input: ListManagedDataIdentifiersInput) async throws -> ListManagedDataIdentifiersOutput
    /// Retrieves information about the accounts that are associated with an Amazon Macie administrator account.
    ///
    /// - Parameter ListMembersInput : [no documentation found]
    ///
    /// - Returns: `ListMembersOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listMembers(input: ListMembersInput) async throws -> ListMembersOutput
    /// Retrieves information about the delegated Amazon Macie administrator account for an organization in Organizations.
    ///
    /// - Parameter ListOrganizationAdminAccountsInput : [no documentation found]
    ///
    /// - Returns: `ListOrganizationAdminAccountsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listOrganizationAdminAccounts(input: ListOrganizationAdminAccountsInput) async throws -> ListOrganizationAdminAccountsOutput
    /// Retrieves information about objects that were selected from an S3 bucket for automated sensitive data discovery.
    ///
    /// - Parameter ListResourceProfileArtifactsInput : [no documentation found]
    ///
    /// - Returns: `ListResourceProfileArtifactsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listResourceProfileArtifacts(input: ListResourceProfileArtifactsInput) async throws -> ListResourceProfileArtifactsOutput
    /// Retrieves information about the types and amount of sensitive data that Amazon Macie found in an S3 bucket.
    ///
    /// - Parameter ListResourceProfileDetectionsInput : [no documentation found]
    ///
    /// - Returns: `ListResourceProfileDetectionsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listResourceProfileDetections(input: ListResourceProfileDetectionsInput) async throws -> ListResourceProfileDetectionsOutput
    /// Retrieves a subset of information about the sensitivity inspection template for an account.
    ///
    /// - Parameter ListSensitivityInspectionTemplatesInput : [no documentation found]
    ///
    /// - Returns: `ListSensitivityInspectionTemplatesOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func listSensitivityInspectionTemplates(input: ListSensitivityInspectionTemplatesInput) async throws -> ListSensitivityInspectionTemplatesOutput
    /// Retrieves the tags (keys and values) that are associated with an Amazon Macie resource.
    ///
    /// - Parameter ListTagsForResourceInput : [no documentation found]
    ///
    /// - Returns: `ListTagsForResourceOutput` : [no documentation found]
    func listTagsForResource(input: ListTagsForResourceInput) async throws -> ListTagsForResourceOutput
    /// Creates or updates the configuration settings for storing data classification results.
    ///
    /// - Parameter PutClassificationExportConfigurationInput : [no documentation found]
    ///
    /// - Returns: `PutClassificationExportConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func putClassificationExportConfiguration(input: PutClassificationExportConfigurationInput) async throws -> PutClassificationExportConfigurationOutput
    /// Updates the configuration settings for publishing findings to Security Hub.
    ///
    /// - Parameter PutFindingsPublicationConfigurationInput : [no documentation found]
    ///
    /// - Returns: `PutFindingsPublicationConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func putFindingsPublicationConfiguration(input: PutFindingsPublicationConfigurationInput) async throws -> PutFindingsPublicationConfigurationOutput
    /// Retrieves (queries) statistical data and other information about Amazon Web Services resources that Amazon Macie monitors and analyzes.
    ///
    /// - Parameter SearchResourcesInput : [no documentation found]
    ///
    /// - Returns: `SearchResourcesOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func searchResources(input: SearchResourcesInput) async throws -> SearchResourcesOutput
    /// Adds or updates one or more tags (keys and values) that are associated with an Amazon Macie resource.
    ///
    /// - Parameter TagResourceInput : [no documentation found]
    ///
    /// - Returns: `TagResourceOutput` : [no documentation found]
    func tagResource(input: TagResourceInput) async throws -> TagResourceOutput
    /// Tests a custom data identifier.
    ///
    /// - Parameter TestCustomDataIdentifierInput : [no documentation found]
    ///
    /// - Returns: `TestCustomDataIdentifierOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func testCustomDataIdentifier(input: TestCustomDataIdentifierInput) async throws -> TestCustomDataIdentifierOutput
    /// Removes one or more tags (keys and values) from an Amazon Macie resource.
    ///
    /// - Parameter UntagResourceInput : [no documentation found]
    ///
    /// - Returns: `UntagResourceOutput` : [no documentation found]
    func untagResource(input: UntagResourceInput) async throws -> UntagResourceOutput
    /// Updates the settings for an allow list.
    ///
    /// - Parameter UpdateAllowListInput : [no documentation found]
    ///
    /// - Returns: `UpdateAllowListOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateAllowList(input: UpdateAllowListInput) async throws -> UpdateAllowListOutput
    /// Enables or disables automated sensitive data discovery for an account.
    ///
    /// - Parameter UpdateAutomatedDiscoveryConfigurationInput : [no documentation found]
    ///
    /// - Returns: `UpdateAutomatedDiscoveryConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateAutomatedDiscoveryConfiguration(input: UpdateAutomatedDiscoveryConfigurationInput) async throws -> UpdateAutomatedDiscoveryConfigurationOutput
    /// Changes the status of a classification job.
    ///
    /// - Parameter UpdateClassificationJobInput : [no documentation found]
    ///
    /// - Returns: `UpdateClassificationJobOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateClassificationJob(input: UpdateClassificationJobInput) async throws -> UpdateClassificationJobOutput
    /// Updates the classification scope settings for an account.
    ///
    /// - Parameter UpdateClassificationScopeInput : [no documentation found]
    ///
    /// - Returns: `UpdateClassificationScopeOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateClassificationScope(input: UpdateClassificationScopeInput) async throws -> UpdateClassificationScopeOutput
    /// Updates the criteria and other settings for a findings filter.
    ///
    /// - Parameter UpdateFindingsFilterInput : [no documentation found]
    ///
    /// - Returns: `UpdateFindingsFilterOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateFindingsFilter(input: UpdateFindingsFilterInput) async throws -> UpdateFindingsFilterOutput
    /// Suspends or re-enables Amazon Macie, or updates the configuration settings for a Macie account.
    ///
    /// - Parameter UpdateMacieSessionInput : [no documentation found]
    ///
    /// - Returns: `UpdateMacieSessionOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateMacieSession(input: UpdateMacieSessionInput) async throws -> UpdateMacieSessionOutput
    /// Enables an Amazon Macie administrator to suspend or re-enable Macie for a member account.
    ///
    /// - Parameter UpdateMemberSessionInput : [no documentation found]
    ///
    /// - Returns: `UpdateMemberSessionOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateMemberSession(input: UpdateMemberSessionInput) async throws -> UpdateMemberSessionOutput
    /// Updates the Amazon Macie configuration settings for an organization in Organizations.
    ///
    /// - Parameter UpdateOrganizationConfigurationInput : [no documentation found]
    ///
    /// - Returns: `UpdateOrganizationConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `ConflictException` : Provides information about an error that occurred due to a versioning conflict for a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateOrganizationConfiguration(input: UpdateOrganizationConfigurationInput) async throws -> UpdateOrganizationConfigurationOutput
    /// Updates the sensitivity score for an S3 bucket.
    ///
    /// - Parameter UpdateResourceProfileInput : [no documentation found]
    ///
    /// - Returns: `UpdateResourceProfileOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateResourceProfile(input: UpdateResourceProfileInput) async throws -> UpdateResourceProfileOutput
    /// Updates the sensitivity scoring settings for an S3 bucket.
    ///
    /// - Parameter UpdateResourceProfileDetectionsInput : [no documentation found]
    ///
    /// - Returns: `UpdateResourceProfileDetectionsOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ServiceQuotaExceededException` : Provides information about an error that occurred due to one or more service quotas for an account.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateResourceProfileDetections(input: UpdateResourceProfileDetectionsInput) async throws -> UpdateResourceProfileDetectionsOutput
    /// Updates the status and configuration settings for retrieving occurrences of sensitive data reported by findings.
    ///
    /// - Parameter UpdateRevealConfigurationInput : [no documentation found]
    ///
    /// - Returns: `UpdateRevealConfigurationOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateRevealConfiguration(input: UpdateRevealConfigurationInput) async throws -> UpdateRevealConfigurationOutput
    /// Updates the settings for the sensitivity inspection template for an account.
    ///
    /// - Parameter UpdateSensitivityInspectionTemplateInput : [no documentation found]
    ///
    /// - Returns: `UpdateSensitivityInspectionTemplateOutput` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : Provides information about an error that occurred due to insufficient access to a specified resource.
    /// - `InternalServerException` : Provides information about an error that occurred due to an unknown internal server error, exception, or failure.
    /// - `ResourceNotFoundException` : Provides information about an error that occurred because a specified resource wasn't found.
    /// - `ThrottlingException` : Provides information about an error that occurred because too many requests were sent during a certain amount of time.
    /// - `ValidationException` : Provides information about an error that occurred due to a syntax error in a request.
    func updateSensitivityInspectionTemplate(input: UpdateSensitivityInspectionTemplateInput) async throws -> UpdateSensitivityInspectionTemplateOutput
}

public enum Macie2ClientTypes {}
